test/__tests__/infer-parser.js

```diff
@@ -0,0 +1,156 @@
+import { runCli } from "../utils";
+
+describe("stdin no path and no parser", () => {
+  describe("logs error and exits with 2", () => {
+    runCli("infer-parser/", [], { input: "foo" }).test({
+      status: 1,
+      stdout: "",
+      write: [],
+    });
+  });
+
+  // TODO (43081j): in prettier, this tests that it exits with 0 for
+  // whatever reason. should we do the same?
```

Comment:
we need to decide what the right behaviour is here

Reply:
I suggest we add a fourth internal only "main" flag/command called "dump" or "echo" or something like that, which we consider enabled when the other 3 ones aren't, and we use that to decide what to output exactly, and we largely align ourselves with Prettier v3.

Reply:
Just mentioning it here that this forth "mode" exists internally now, I called it "dump".